In [172]:
# Dependencies and Setup
import pandas as pd

# File to Load (Remember to Change These)
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)

In [173]:
purchase_data.head(10)

Purchase ID             SN  Age                 Gender  Item ID  \
0            0        Lisim78   20                   Male      108   
1            1    Lisovynya38   40                   Male      143   
2            2     Ithergue48   24                   Male       92   
3            3  Chamassasya86   24                   Male      100   
4            4      Iskosia90   23                   Male      131   
5            5        Yalae81   22                   Male       81   
6            6      Itheria73   36                   Male      169   
7            7    Iskjaskst81   20                   Male      162   
8            8      Undjask33   22                   Male       21   
9            9    Chanosian48   35  Other / Non-Disclosed      136   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44  
5                                  Dreamkiss   3.61  
6     Interrogator, Blood Blade of the Queen   2.18  
7                              Abyssal Shard   2.67  
8                                  Souleater   1.10  
9               Ghastly Adamantite Protector   3.58

In [174]:
#Finding number of unique players

player_demographics = purchase_data.loc[:,['Gender','SN','Age']]
player_demographics = player_demographics.drop_duplicates()
tot_players=player_demographics.count()[0]
tot_players_df = pd.DataFrame({'Total Players': [tot_players]})
tot_players_df

Total Players
0            576

In [175]:
#Purchasing Analysis
#Finding numver of unique items, their average price, number of purchases,and total revenue

unique_item_count= len(purchase_data['Item ID'].unique())
avg_price = round(float(purchase_data['Price'].mean()),2)
purchases_count = len(purchase_data['Price'])
price_sum = float(purchase_data['Price'].sum())

item_df = pd.DataFrame({
    'Number of Unique Items': [unique_item_count],
    'Average Price': '$'+str(avg_price),
    'Number of Purchases': [purchases_count],
    'Total Revenue': '$'+str(price_sum)
})
item_df

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     179         $3.05                  780      $2379.77

In [176]:
#Gender Demographics
#Find the percentage and count of all of the genders of the players 

gender_df = purchase_data[['Gender','SN']].drop_duplicates(subset = 'SN')
gender_count = gender_df['Gender'].value_counts()
gender_total = gender_count.sum()
gender_percent = round((gender_count/gender_total)*100,2)

gender_count_df = pd.DataFrame(gender_count)
gender_percent_df = pd.DataFrame(gender_percent)

gender_demographics_df = gender_count_df.merge(gender_percent_df,left_index = True, right_index = True)
gender_demographics_df.columns=['Total Count','Percentage of Players']
gender_demographics_df['Total Count'] = gender_demographics_df['Total Count'].map('${:,.2f}'.format)
gender_demographics_df['Percentage of Players'] = gender_demographics_df['Percentage of Players'].map('{:,.2f}%'.format)
gender_demographics_df

Total Count Percentage of Players
Male                      $484.00                84.03%
Female                     $81.00                14.06%
Other / Non-Disclosed      $11.00                 1.91%

In [177]:
#Purchasing Analysis by Gender
#Find the purchase count, avg. purchase price, purchase total per person by gender

gender_df2 = purchase_data.groupby('Gender')

purchase_count = round(gender_df2['Purchase ID'].count(),0)
avg_purchase_price = round(gender_df2['Price'].mean(),2)
total_value = round(gender_df2['Price'].sum(),2)
purchase_gender = round(total_value/gender_count,2)

analysis_df = pd.DataFrame([purchase_count,avg_purchase_price,total_value,purchase_gender])
purchasing_analysis_gender_df = analysis_df.T
purchasing_analysis_gender_df.columns = ['Purchase Count','Average Purchase Price','Total Purchase Value','Average Total Purchase per Gender']
purchasing_analysis_gender_df['Average Purchase Price'] = purchasing_analysis_gender_df['Average Purchase Price'].map('${:,.2f}'.format)
purchasing_analysis_gender_df['Total Purchase Value'] = purchasing_analysis_gender_df['Total Purchase Value'].map('${:,.2f}'.format)
purchasing_analysis_gender_df['Average Total Purchase per Gender'] = purchasing_analysis_gender_df['Average Total Purchase per Gender'].map('${:,.2f}'.format)
purchasing_analysis_gender_df

Purchase Count Average Purchase Price  \
Gender                                                         
Female                          113.0                  $3.20   
Male                            652.0                  $3.02   
Other / Non-Disclosed            15.0                  $3.35   

                      Total Purchase Value Average Total Purchase per Gender  
Gender                                                                        
Female                             $361.94                             $4.47  
Male                             $1,967.64                             $4.07  
Other / Non-Disclosed               $50.19                             $4.56

In [178]:
#Age Demographics
#Find the total count and what percent of the total players by age group
bins = [0,9,14,19,24,29,34,39,100]
bin_labels = ['<10','10-14','15-19','20-24','25-29','30-34','35-39','40+']
purchase_data['Age Group'] = pd.cut(purchase_data['Age'],bins,labels=bin_labels)
unique_age_df = purchase_data[['Age Group','SN']].drop_duplicates(subset='SN')

age_df = unique_age_df.groupby('Age Group').count()
age_counts = age_df['SN']
age_percentages = round(age_counts/tot_players*100,2)

age_df2 = pd.DataFrame([age_counts,age_percentages])
age_demographics_df = age_df2.T
age_demographics_df.columns = ['Total Count','Percentage of Players']
age_demographics_df['Percentage of Players'] = age_demographics_df['Percentage of Players'].map('{:,.2f}%'.format)
age_demographics_df

Total Count Percentage of Players
Age Group                                   
<10               17.0                 2.95%
10-14             22.0                 3.82%
15-19            107.0                18.58%
20-24            258.0                44.79%
25-29             77.0                13.37%
30-34             52.0                 9.03%
35-39             31.0                 5.38%
40+               12.0                 2.08%

In [179]:
#Purchasing Analysis by Age
#Find the purchase count, avg. purchase price, purchase total per person by age group
#the bins were created in the previous cell

reduced_rows = purchase_data[['Age Group','SN','Price']]
reduced_groupby = reduced_rows.groupby('Age Group')
purchase_count = reduced_groupby['Price'].count()
avg_price = round(reduced_groupby['Price'].mean(),2)
total_spent = round(reduced_groupby['Price'].sum(),2)
spending_per_person = round(total_spent/age_counts,2)

analysis_age = pd.DataFrame([purchase_count,avg_price,total_spent,spending_per_person])
purchasing_analysis_age_df = analysis_age.T
purchasing_analysis_age_df.columns = ['Purchase Count', 'Average Purchase Price', 'Total Purchase Value','Avg Total Purchase per Person']
purchasing_analysis_age_df['Average Purchase Price'] = purchasing_analysis_age_df['Average Purchase Price'].map('${:,.2f}'.format)
purchasing_analysis_age_df['Total Purchase Value'] = purchasing_analysis_age_df['Total Purchase Value'].map('${:,.2f}'.format)
purchasing_analysis_age_df['Avg Total Purchase per Person'] = purchasing_analysis_age_df['Avg Total Purchase per Person'].map('${:,.2f}'.format)
purchasing_analysis_age_df

Purchase Count Average Purchase Price Total Purchase Value  \
Age Group                                                               
<10                  23.0                  $3.35               $77.13   
10-14                28.0                  $2.96               $82.78   
15-19               136.0                  $3.04              $412.89   
20-24               365.0                  $3.05            $1,114.06   
25-29               101.0                  $2.90              $293.00   
30-34                73.0                  $2.93              $214.00   
35-39                41.0                  $3.60              $147.67   
40+                  13.0                  $2.94               $38.24   

          Avg Total Purchase per Person  
Age Group                                
<10                               $4.54  
10-14                             $3.76  
15-19                             $3.86  
20-24                             $4.32  
25-29                             $3.81  
30-34                             $4.12  
35-39                             $4.76  
40+                               $3.19

In [180]:
#Top Spender
#sort through data frame to find the player that spent the most in the game

SN_df = purchase_data.groupby('SN')
purchase_count = SN_df['Gender'].count()
avg_spending = round(SN_df['Price'].mean(),2)
tot_price = round(SN_df['Price'].sum(),2)

analysis_price = pd.DataFrame([purchase_count,avg_spending,tot_price])
top_spender_analysis_df = analysis_price.T
top_spender_analysis_df.columns = ['Purchase Count','Average Purchase Price','Total Purchase Value']
top_spender_analysis_df = top_spender_analysis_df.sort_values('Total Purchase Value',ascending=False).reset_index()
top_spender_analysis_df['Average Purchase Price'] = top_spender_analysis_df['Average Purchase Price'].map('${:,.2f}'.format)
top_spender_analysis_df['Total Purchase Value'] = top_spender_analysis_df['Total Purchase Value'].map('${:,.2f}'.format)
top_spender_analysis_df.head()

SN  Purchase Count Average Purchase Price Total Purchase Value
0    Lisosia93             5.0                  $3.79               $18.96
1  Idastidru52             4.0                  $3.86               $15.45
2   Chamjask73             3.0                  $4.61               $13.83
3       Iral74             4.0                  $3.40               $13.62
4  Iskadarya95             3.0                  $4.37               $13.10

In [181]:
#Most Popular Items
#Sort through the data to find the most popular items purchased

items_df = purchase_data.groupby(['Item ID','Item Name'])
purchase_count = items_df['Gender'].count()
avg_spending = round(items_df['Price'].mean(),2)
tot_purchase = round(items_df['Price'].sum(),2)

analysis_items = pd.DataFrame([purchase_count,avg_spending,tot_purchase])
most_popular_items_df = analysis_items.T
place_holder = analysis_items.T
most_popular_items_df.columns = ['Purchase Count','Item Price','Total Purchase Value']
most_popular_items_df = most_popular_items_df.sort_values('Purchase Count',ascending=False)
most_popular_items_df['Item Price'] = most_popular_items_df['Item Price'].map('${:,.2f}'.format)
most_popular_items_df['Total Purchase Value'] = most_popular_items_df['Total Purchase Value'].map('${:,.2f}'.format)
most_popular_items_df.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                            13.0   
178     Oathbreaker, Last Hope of the Breaking Storm            12.0   
145     Fiery Glass Crusader                                     9.0   
132     Persuasion                                               9.0   
108     Extraction, Quickblade Of Trembling Hands                9.0   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

In [182]:
#Most Profitable Items
#sort throught the data to find the item with the maximum total purchase value

place_holder.columns = ['Purchase Count','Item Price','Total Purchase Value']
most_profitable_df = place_holder.sort_values('Total Purchase Value', ascending=False)
most_profitable_df['Item Price'] = most_profitable_df['Item Price'].map('$ {:,.2f}'.format)
most_profitable_df['Total Purchase Value'] = most_profitable_df['Total Purchase Value'].map('${:,.2f}'.format)
most_profitable_df.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                            13.0   
178     Oathbreaker, Last Hope of the Breaking Storm            12.0   
82      Nirvana                                                  9.0   
145     Fiery Glass Crusader                                     9.0   
103     Singed Scalpel                                           8.0   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                     $ 4.61   
178     Oathbreaker, Last Hope of the Breaking Storm     $ 4.23   
82      Nirvana                                          $ 4.90   
145     Fiery Glass Crusader                             $ 4.58   
103     Singed Scalpel                                   $ 4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80

**Observable Trends Based on the Data**

1. Most of the people who play this game are males, making up 84.03% of the total players.
2. Most of the players of this game are between the ages of 20-24, making up 44.79% of the total players. Almost all of the players are below the age of 40: only 2.08% of the players are above the age of 40.
3. Even though an purchased item may per popular (has a large number of purchases), it may not be the most profitable. Persuasion was the 3rd most purchased item, but it did not even make the top 5 for the most profitable